src/test/scala/ai/privado/languageEngine/go/tagger/sink/GoAPITaggerTest.scala

```diff
@@ -0,0 +1,123 @@
+import ai.privado.model.*
+import io.shiftleft.semanticcpg.language.*
+
+class GoAPITaggerTest extends GoTaggingTestBase {
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+    new IdentifierTagger(cpg, ruleCache, taggerCache).createAndApply()
+    new GoAPITagger(cpg, ruleCache, privadoInput).createAndApply()
+  }
+
+  override val goFileContents =
+    """
+      | package main
+      |
+      |import (
+      |	"bytes"
+      |	"encoding/json"
+      |	"fmt"
+      |	"net/http"
+      |)
+      |
+      |type User struct {
+      |	FirstName     string
+      |	Age      int
+      |	Location string
+      |	Email    string
+      |}
+      |func (client *APIClient) SendUser(user User) error {
+      |	http_url := client.BaseURL + "/users"
+      |	payload, err := json.Marshal(user)
+      |	if err != nil {
+      |		return err
+      |	}
+      |
+      |	resp, err := http.Post(http_url, "application/json", bytes.NewBuffer(payload))
+      |	if err != nil {
+      |		return err
+      |	}
+      |	defer resp.Body.Close()
+      |
+      |	if resp.StatusCode != http.StatusOK {
+      |		return fmt.Errorf("failed to send user: %s", resp.Status)
+      |	}
+      |
+      |	fmt.Println("User sent successfully!")
+      |	return nil
+      |}
+      |
+      |func main() {
+      |	// Create a new User object
+      |	user := User{
+      |		Name:     "John Doe",
+      |		Age:      25,
+      |		Location: "New York",
+      |		Email: "abc@gmail.com",
+      |	}
+      |
+      |	// Create a new API client
+      |	client := APIClient{
+      |		BaseURL: "https://api.example.com",
+      |	}
+      |
+      |	// Send the User object to the API client
+      |	err := client.SendUser(user)
+      |	if err != nil {
+      |		fmt.Printf("Error sending user: %s\n", err.Error())
+      |	}
+      |}
+      |
+      |""".stripMargin
+
+  "Tagging api sink" should {
+    "check tag of api sink" in {
+      val identifierNodes = cpg.member("FirstName").tag.nameExact(Constants.id).l
+      identifierNodes.size shouldBe 1
+      identifierNodes.value.head shouldBe "Data.Sensitive.FirstName"
+
+      val List(postCallNode) = cpg.call("Post").l
+
+      postCallNode.tag.size shouldBe 6
+      postCallNode.tag.nameExact("id").value.head shouldBe "Sinks.ThirdParties.API"
+      postCallNode.tag.nameExact("nodeType").value.head shouldBe "api"
+      postCallNode.tag.nameExact("catLevelOne").value.head shouldBe "sinks"
+      postCallNode.tag.nameExact("catLevelTwo").value.head shouldBe "third_parties"
+      postCallNode.tag.nameExact("third_partiesapi").value.head shouldBe "Sinks.ThirdParties.API"
+      postCallNode.tag.nameExact("apiUrlSinks.ThirdParties.API").value.head shouldBe "API"
+      // TODO: check tags on postCallNode for below cases
+      // 1. When literal holds some url
+      // 2. When Identifier is matching with apiIdentifier pattern
+      // 3. When nothing is matching it should tagged with "API"
```

Comment:
Better to add these test cases as well, if a test case is not working that can be marked ignore for now

Reply:
Added ignore case: https://github.com/Privado-Inc/privado-core/pull/779/commits/84ee8e91ff77b8739028ffce9a692fce2737fe46